src/lib/mdns/TxtFields.cpp

```diff
@@ -206,40 +220,58 @@ TxtFieldKey GetTxtFieldKey(const ByteSpan & key)
 
 } // namespace Internal
 
-void FillNodeDataFromTxt(const ByteSpan & key, const ByteSpan & val, DiscoveredNodeData * nodeData)
+void FillNodeDataFromTxt(const ByteSpan & key, const ByteSpan & val, DiscoveredNodeData & nodeData)
 {
     Internal::TxtFieldKey keyType = Internal::GetTxtFieldKey(key);
     switch (keyType)
     {
     case Internal::TxtFieldKey::kLongDiscriminator:
-        nodeData->longDiscriminator = Internal::GetLongDisriminator(val);
+        nodeData.longDiscriminator = Internal::GetLongDisriminator(val);
         break;
     case Internal::TxtFieldKey::kVendorProduct:
-        nodeData->vendorId  = Internal::GetVendor(val);
-        nodeData->productId = Internal::GetProduct(val);
+        nodeData.vendorId  = Internal::GetVendor(val);
+        nodeData.productId = Internal::GetProduct(val);
         break;
     case Internal::TxtFieldKey::kAdditionalPairing:
-        nodeData->additionalPairing = Internal::GetAdditionalPairing(val);
+        nodeData.additionalPairing = Internal::GetAdditionalPairing(val);
         break;
     case Internal::TxtFieldKey::kCommissioningMode:
-        nodeData->commissioningMode = Internal::GetCommissioningMode(val);
+        nodeData.commissioningMode = Internal::GetCommissioningMode(val);
         break;
     case Internal::TxtFieldKey::kDeviceType:
-        nodeData->deviceType = Internal::GetDeviceType(val);
+        nodeData.deviceType = Internal::GetDeviceType(val);
         break;
     case Internal::TxtFieldKey::kDeviceName:
-        Internal::GetDeviceName(val, nodeData->deviceName);
+        Internal::GetDeviceName(val, nodeData.deviceName);
         break;
     case Internal::TxtFieldKey::kRotatingDeviceId:
-        Internal::GetRotatingDeviceId(val, nodeData->rotatingId, &nodeData->rotatingIdLen);
+        Internal::GetRotatingDeviceId(val, nodeData.rotatingId, &nodeData.rotatingIdLen);
         break;
     case Internal::TxtFieldKey::kPairingInstruction:
-        Internal::GetPairingInstruction(val, nodeData->pairingInstruction);
+        Internal::GetPairingInstruction(val, nodeData.pairingInstruction);
         break;
     case Internal::TxtFieldKey::kPairingHint:
-        nodeData->pairingHint = Internal::GetPairingHint(val);
+        nodeData.pairingHint = Internal::GetPairingHint(val);
+        break;
+    default:
+        break;
+    }
+}
+
+void FillNodeDataFromTxt(const ByteSpan & key, const ByteSpan & value, ResolvedNodeData & nodeData)
+{
+    switch (Internal::GetTxtFieldKey(key))
+    {
+    case Internal::TxtFieldKey::kMrpRetryIntervalIdle:
+        nodeData.mMrpRetryIntervalIdle = Internal::MakeU32FromAsciiDecimal(value);
```

Comment:
Shouldn't we enforce the "no more than 3600000" requirement somewhere?  We could do it in higher layers, I guess, but doing it here would be nice in terms of making sure it can't get messed up.

Same for `kMrpRetryIntervalActive`.

Reply:
Right, I tried to be consistent with existing TXT entries (for commissionable nodes) for which there's no validation at the DNS level, but I agree it would make more sense to validate data as early as possible.